File: app/src/main/java/com/cappielloantonio/play/subsonic/models/SearchResult2.java
package com.cappielloantonio.play.subsonic.models;

import com.tickaroo.tikxml.annotation.Element;
import com.tickaroo.tikxml.annotation.Xml;

import java.util.ArrayList;
import java.util.List;

@Xml
public class SearchResult2 {
    @Element(name = "artist")
    protected List<Artist> artists;
    @Element(name = "album")
    protected List<Child> albums;
    @Element(name = "song")
    protected List<Child> songs;

    public List<Artist> getArtists() {
        if (artists == null) {
            artists = new ArrayList<>();
        }
        return this.artists;
    }

    public void setArtists(List<Artist> artists) {
        this.artists = artists;
    }

    public List<Child> getAlbums() {
        if (albums == null) {
            albums = new ArrayList<>();
        }
        return this.albums;
    }

    public void setAlbums(List<Child> albums) {
        this.albums = albums;
    }

    public List<Child> getSongs() {
        if (songs == null) {
            songs = new ArrayList<>();
        }
        return this.songs;
    }

    public void setSongs(List<Child> songs) {
        this.songs = songs;
    }
}

File: app/src/main/java/com/cappielloantonio/tempo/subsonic/models/SearchResult2.kt
package com.cappielloantonio.tempo.subsonic.models

import androidx.annotation.Keep
import com.google.gson.annotations.SerializedName

@Keep
class SearchResult2 {
    @SerializedName("artist")
    var artists: List<Artist>? = null
    @SerializedName("album")
    var albums: List<Child>? = null
    @SerializedName("song")
    var songs: List<Child>? = null
}